chore(build): Update ndm dockerfiles to use ndm build base image

z0marlin opened the pull request:
Build ndm images using the [ndm build base image](https://github.com/openebs/openSeaChest/pkgs/container/ndm-build-base)

**Why is this PR required? What issue does it fix?**:
The ndm docker images built during CI install dependencies (openSeachest, libudev, libblkid) again and again for each build. This increases the duration of the CI run. This PR aims to make the CI runs faster.

**What this PR does?**:
Uses a custom base image for ndm builds with all the build dependencies pre-installed.

**Checklist:**
- [x] Fixes #628 
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated? (NA)
- [ ] Commit has unit tests (NA)
- [ ] Commit has integration tests (NA)

z0marlin committed:
chore(build): Update ndm dockerfiles to use ndm build base image

Signed-off-by: Aditya Jain <aditya.jainadityajain.jain@gmail.com>

akhilerm commented on build/ndm-daemonset/Dockerfile:
```diff
@@ -14,7 +14,7 @@
 # This Dockerfile builds node-disk-manager daemon
 #
-FROM golang:1.14.7 as build
+FROM ghcr.io/openebs/ndm-build-base:20211125 as build
```
```suggestion
FROM ghcr.io/openebs/ndm-build-base:20211202 as build
```

akhilerm commented on build/ndm-exporter/Dockerfile:
```diff
@@ -14,7 +14,7 @@
 # This Dockerfile builds ndm exporter
 #
-FROM golang:1.14.7 as build
+FROM ghcr.io/openebs/ndm-build-base:20211125 as build
```
```suggestion
FROM ghcr.io/openebs/ndm-build-base:20211202 as build
```

akhilerm commented on build/ndm-operator/Dockerfile:
```diff
@@ -14,7 +14,7 @@
 # This Dockerfile builds ndm operator
 #
-FROM golang:1.14.7 as build
+FROM ghcr.io/openebs/ndm-build-base:20211125 as build
```
```suggestion
FROM ghcr.io/openebs/ndm-build-base:20211202 as build
```

akhilerm reviewed (commented):
Need to update the base image versions

akhilerm committed:
Update build/ndm-daemonset/Dockerfile

akhilerm committed:
Update build/ndm-exporter/Dockerfile

akhilerm committed:
Update build/ndm-operator/Dockerfile

akhilerm reviewed (dismissed):
LGTM

z0marlin committed:
Update seachest cgo build CFLAGS

Signed-off-by: Aditya Jain <aditya.jainadityajain.jain@gmail.com>

z0marlin committed:
Remove redundant dependencies from ndm dockerfiles

The ndm build base image is shipped with the packages make and git.
Remove redundant installation.

Signed-off-by: Aditya Jain <aditya.jainadityajain.jain@gmail.com>

akhilerm reviewed (approved):
LGTM
